File: app/src/test/java/openfoodfacts/github/scrachx/openfood/models/NutrientLevelsTest.java
package openfoodfacts.github.scrachx.openfood.models;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

import org.junit.Before;
import org.junit.Test;

import static com.google.common.truth.Truth.assertThat;

/**
 * @author herau
 */
public class NutrientLevelsTest {
    private NutrientLevels nutrientLevels;

    @Before
    public void setUp() {
        nutrientLevels = new NutrientLevels();
        nutrientLevels.setFat(NutrimentLevel.LOW);
        nutrientLevels.setSalt(NutrimentLevel.MODERATE);
        nutrientLevels.setSaturatedFat(NutrimentLevel.HIGH);
        nutrientLevels.setSugars(NutrimentLevel.MODERATE);
    }

    @Test
    public void jsonSerialization_ok() {
        JsonNode jsonNode = new ObjectMapper().valueToTree(nutrientLevels);

        assertThat(jsonNode.get("fat").asText()).isEqualTo(NutrimentLevel.LOW.toString());
        assertThat(jsonNode.get("salt").asText()).isEqualTo(NutrimentLevel.MODERATE.toString());
        assertThat(jsonNode.get("saturated-fat").asText()).isEqualTo(NutrimentLevel.HIGH.toString());
        assertThat(jsonNode.get("sugars").asText()).isEqualTo(NutrimentLevel.MODERATE.toString());
    }

    @Test
    public void jsonDeserialization_ok() throws Exception {
        ObjectMapper objectMapper = new ObjectMapper();
        NutrientLevels nutrientLevelsResult = objectMapper.treeToValue(
            objectMapper.createObjectNode()
                .put("fat", "low")
                .put("salt", "moderate")
                .put("saturated-fat", "high")
                .put("sugars", "moderate"), NutrientLevels.class);

        assertThat(nutrientLevelsResult.getFat()).isEqualTo(nutrientLevels.getFat());
        assertThat(nutrientLevelsResult.getSugars()).isEqualTo(nutrientLevels.getSugars());
        assertThat(nutrientLevelsResult.getSaturatedFat()).isEqualTo(nutrientLevels.getSaturatedFat());
        assertThat(nutrientLevelsResult.getSalt()).isEqualTo(nutrientLevels.getSalt());
    }
}

File: app/src/test/java/openfoodfacts/github/scrachx/openfood/models/NutrientLevelsTest.kt
package openfoodfacts.github.scrachx.openfood.models

import com.fasterxml.jackson.databind.JsonNode
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.google.common.truth.Truth
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

/**
 * @author herau
 */
class NutrientLevelsTest {
    private lateinit var nutrientLevels: NutrientLevels

    @BeforeEach
    fun setUp() {
        nutrientLevels = NutrientLevels()
        nutrientLevels.fat = NutrimentLevel.LOW
        nutrientLevels.salt = NutrimentLevel.MODERATE
        nutrientLevels.saturatedFat = NutrimentLevel.HIGH
        nutrientLevels.sugars = NutrimentLevel.MODERATE
    }

    @Test
    fun jsonSerialization_ok() {
        val jsonNode = jacksonObjectMapper().valueToTree<JsonNode>(nutrientLevels)
        Truth.assertThat(jsonNode["fat"].asText()).isEqualTo(NutrimentLevel.LOW.toString())
        Truth.assertThat(jsonNode["salt"].asText()).isEqualTo(NutrimentLevel.MODERATE.toString())
        Truth.assertThat(jsonNode["saturated-fat"].asText()).isEqualTo(NutrimentLevel.HIGH.toString())
        Truth.assertThat(jsonNode["sugars"].asText()).isEqualTo(NutrimentLevel.MODERATE.toString())
    }

    @Test
    fun jsonDeserialization_ok() {
        val objectMapper = jacksonObjectMapper()
        val nutrientLevelsResult = objectMapper.treeToValue(
                objectMapper.createObjectNode()
                        .put("fat", "low")
                        .put("salt", "moderate")
                        .put("saturated-fat", "high")
                        .put("sugars", "moderate"), NutrientLevels::class.java)
        Truth.assertThat(nutrientLevelsResult.fat).isEqualTo(nutrientLevels.fat)
        Truth.assertThat(nutrientLevelsResult.sugars).isEqualTo(nutrientLevels.sugars)
        Truth.assertThat(nutrientLevelsResult.saturatedFat).isEqualTo(nutrientLevels.saturatedFat)
        Truth.assertThat(nutrientLevelsResult.salt).isEqualTo(nutrientLevels.salt)
    }
}